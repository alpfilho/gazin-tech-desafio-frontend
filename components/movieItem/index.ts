export { MovieItem } from './movieItem';
export type { MovieItemI } from './movieItem';
