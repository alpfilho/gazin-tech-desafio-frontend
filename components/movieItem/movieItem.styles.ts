import styled from 'styled-components';

export const MovieItemContainer = styled.a`
	display: flex;
	flex-direction: column;

	width: 216px;
	height: auto;

	border-radius: 1.5em;
	overflow: hidden;

	background-color: ${({ theme }) => theme.background2};
`;

export const ImageContainer = styled.div`
	position: relative;
	width: 100%;
	padding: 0.5em;
`;

export const TextContainer = styled.div`
	padding: 0.25em 1em 1em 1em;
	color: ${({ theme }) => theme.text};
`;

export const Title = styled.div`
	font-family: 'Poppins', sans-serif;
	font-weight: 700;
	font-size: 1.125em;

	overflow: hidden;
	white-space: nowrap;
	text-overflow: ellipsis;

	color: ${({ theme }) => theme.text};
`;

export const DateContainer = styled.div`
	font-family: 'Roboto', sans-serif;
	font-size: 0.75em;
`;
