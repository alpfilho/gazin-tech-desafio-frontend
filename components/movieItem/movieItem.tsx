import React, { useCallback } from 'react';

import {
	MovieItemContainer,
	ImageContainer,
	TextContainer,
	Title,
	DateContainer
} from './movieItem.styles';

export interface MovieItemI {
	title: string;
	image: string;
	releaseDate: string;
}

export const MovieItem: React.FC<MovieItemI> = ({ title, image, releaseDate }) => {
	const getYear = useCallback((releaseDate) => {
		const data = new Date(releaseDate);
		return data.getFullYear();
	}, []);

	return (
		<MovieItemContainer>
			<ImageContainer>
				<img src={image} alt="" />
			</ImageContainer>
			<TextContainer>
				<Title>{title}</Title>
				<DateContainer>{getYear(releaseDate)}</DateContainer>
			</TextContainer>
		</MovieItemContainer>
	);
};
