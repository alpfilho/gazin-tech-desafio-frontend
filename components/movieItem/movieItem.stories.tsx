import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { MovieItem } from './movieItem';

export default {
	title: 'Movie/Item',
	component: MovieItem
} as ComponentMeta<typeof MovieItem>;

export const Default: ComponentStory<typeof MovieItem> = (args) => <MovieItem {...args} />;

Default.args = {
	id: 230896,
	image: 'http://image.tmdb.org/t/p/w500/j2mv9DczY12TZtA8L8PcMqaiSkI.jpg',
	title: 'Homem de Ferro e Hulk - Super-Her√≥is Unidos',
	releaseDate: '2013-12-03'
};
