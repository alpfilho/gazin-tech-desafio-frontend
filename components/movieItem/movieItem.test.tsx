import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './movieItem.stories';

const { Default } = composeStories(stories);

describe('MovieItem', () => {
	it('render a movie item', () => {
		render(<Default />);

		const movieTitle = screen.getByText('Homem de Ferro e Hulk - Super-HerÃ³is Unidos');
		const movieRelease = screen.getByText('2013');
		expect(movieTitle).toBeInTheDocument();
		expect(movieRelease).toBeInTheDocument();
	});

	it('movie item render unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
