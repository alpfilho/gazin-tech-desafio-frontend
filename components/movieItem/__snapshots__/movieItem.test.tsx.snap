// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem movie item render unchanged 1`] = `
<div>
  <a
    class="MovieItemContainer-sc-11o4gv2-0 qpfDR"
    href="/movie-detail/230896"
  >
    <div
      class="ImageContainer-sc-11o4gv2-1 cXNnZm"
    >
      <div
        class="MoviePosterContainer-sc-i25oql-0 eyVCLd"
      >
        <div
          class="PosterContent-sc-i25oql-1 eiTosH"
        >
          <div
            class="MovieImage-sc-i25oql-2 ja-dEVy"
          >
            <div
              class="MovieImageContent-sc-i25oql-3 fokrKX"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt=""
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center center;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="TextContainer-sc-11o4gv2-2 TnVlx"
    >
      <div
        class="Title-sc-11o4gv2-3 jgkVYj"
        title="Homem de Ferro e Hulk - Super-Heróis Unidos"
      >
        Homem de Ferro e Hulk - Super-Heróis Unidos
      </div>
      <div
        class="DateContainer-sc-11o4gv2-4 dFJNdF"
      >
        2013
      </div>
    </div>
  </a>
</div>
`;
